app/src/main/java/com/juniorkekod/disneyplusclonekekod/presentation/tracing/models/DownloadedSeries.kt

```diff
@@ -0,0 +1,12 @@
+package com.juniorkekod.disneyplusclonekekod.presentation.tracing.models
+
+import android.graphics.drawable.Drawable
+
+data class DownloadedSeries(
+    override val image: Drawable? = null,
+    override val name: String? = null,
+    override val size: Int? = null,
+    override val watchProgress: Int? = null,
+    override val time: Int? = null,
+    val episodeNumber: Int? = null
+) : DownloadableVideos()
```

Comment:
Her iki data classta'da tekrar tekrar override edeceksek abstract bir class oluşturup data class'ı burdan türetmenin bir mantığı yok. Zaten kalıtım aldığı yerden o değişkenlere ulaşılabilecek. Data class içerisinde override yapısı çok fazla kullanılan bir yol değil. Her data class değişkenlerini kendine özgü tutabilir bunun için kalıtım almamalısın. Base bir data class oluşturmak istersen örneğin bir response model için: success değeri response_code vs gibi her response modelden dönecek değerleri oraya alabilirsin. Ama onu da abstract yapmamalısın.